pkg/cmd/admin/top/images.go

```diff
@@ -0,0 +1,226 @@
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+)
+
+const (
+	TopImagesRecommendedName = "images"
+
+	topImagesLong = `Show usage statistics for Images
+
+This command analyzes all the Images managed by the platform and presents current
+usage statistics.`
+
+	topImagesExample = `  # Show usage statistics for Images
+  %[1]s %[2]s`
+)
+
+// NewCmdTopImages implements the OpenShift cli top images command.
+func NewCmdTopImages(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	opts := &TopImagesOptions{}
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   "Show usage statistics for Images",
+		Long:    topImagesLong,
+		Example: fmt.Sprintf(topImagesExample, parentName, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(opts.Complete(f, cmd, args, out))
+			kcmdutil.CheckErr(opts.Validate(cmd))
+			kcmdutil.CheckErr(opts.Run())
+		},
+	}
+
+	return cmd
+}
+
+type TopImagesOptions struct {
+	// internal values
+	Images  *imageapi.ImageList
+	Streams *imageapi.ImageStreamList
+	Pods    *kapi.PodList
+
+	// helpers
+	out      io.Writer
+	osClient client.Interface
+	kClient  kclient.Interface
+}
+
+// Complete turns a partially defined TopImagesOptions into a solvent structure
+// which can be validated and used for showing limits usage.
+func (o *TopImagesOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	osClient, kClient, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	namespace := cmd.Flag("namespace").Value.String()
+	if len(namespace) == 0 {
+		namespace = kapi.NamespaceAll
+	}
+	o.out = out
+
+	allImages, err := osClient.Images().List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Images = allImages
+
+	allStreams, err := osClient.ImageStreams(namespace).List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Streams = allStreams
+
+	allPods, err := kClient.Pods(namespace).List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Pods = allPods
+
+	return nil
+}
+
+// Validate ensures that a TopImagesOptions is valid and can be used to execute command.
+func (o TopImagesOptions) Validate(cmd *cobra.Command) error {
+	return nil
+}
+
+// Run contains all the necessary functionality to show current image references.
+func (o TopImagesOptions) Run() error {
+	infos := o.imagesTop()
+	Print(o.out, ImageColumns, infos)
+	return nil
+}
+
+var ImageColumns = []string{"NAME", "IMAGESTREAMTAG", "PARENTS", "USAGE", "METADATA", "STORAGE"}
+
+// imageInfo contains statistic information about Image usage.
+type imageInfo struct {
+	Image          string
+	ImageStreamTag string
```

Comment:
You could truncate the array with ellipsis if there's more then three image stream tags per image.